### PR TITLE
Adapt the web UI to use the Agama network service

### PR DESCRIPTION
**Pull request by imobachgs**
## Problem

The web UI still talks directly to NetworkManager to set up network connections.

## Solution

This is the first set of changes aimed to stop using the NetworkManager service from the web UI. Instead, it should talk to Agama's network service (which relies on NetworkManager itself).  

This pull request adapts the web UI to use Agama's network service to write the changes (including connecting to wireless devices). However, the status information is still retrieved from NetworkManager because those bits are missing from our API.

During the process, we needed to introduce a few changes to Agama's network service:

* Use UUIDs to identify connections instead of the Id (the name).
* Write only changed network connections.
* Rollback to the previous state if some connection cannot be updated.
* Better error handling at D-Bus level.

## Future

* Improve (almost non-existent) error handling.
* Retrieve the status information from Agama's network service.

## Testing

- *Adapted unit tests*
- *Tested manually*

### CODE REVIEW

File: web/src/client/network/index.js [hunk 1]
```diff
@@ -156,48 +187,204 @@ o  *   NetworkManagerAdapter.
...
   /**
    * Adds a new connection
    *
+   * If a connection with the given ID already exists, it updates such a
+   * connection.
+   *
    * @param {Connection} connection - Connection to add
+   * @return {Promise<Connection>} the added connection
    */
   async addConnection(connection) {
```
Comment (joseivanlopez) [inline web/src/client/network/index.js:214]: NP: What do think about renaming this method to `AddOrUpdateConnection` ?

File: web/src/client/network/index.js [hunk 2]
```diff
@@ -156,48 +187,204 @@ o  *   NetworkManagerAdapter.
...
 
   /**
    * Updates the connection
    *
-   * It uses the 'path' to match the connection in the backend.
+   * It uses the 'uuid' to match the connection in the backend.
    *
    * @param {Connection} connection - Connection to update
+   * @return {Promise<boolean>} - the promise resolves to true if the connection
+   *   was successfully updated and to false it it does not exist.
```
Comment (joseivanlopez) [inline web/src/client/network/index.js:328]: Typo: it it

File: web/src/client/network/model.js [hunk 3]
```diff
@@ -60,6 +60,17 @@ const SecurityProtocols = Object.freeze({
   _8021X: "802.1X"
 });
 
+// security protocols
+// const AgamaSecurityProtocols = Object.freeze({
```
Comment (joseivanlopez) [inline web/src/client/network/model.js:64]: Leftover?

**Review by joseivanlopez — COMMENTED** [review]
Looks great! Only minor comments.

File: web/src/client/network/index.js (hunk 1 above)
Comment (imobachgs) [reply web/src/client/network/index.js:214]: I thought about that, but it is weird having an `addOrUpdateConnection` and `updateConnection.` Perhaps we might need to merge them and have something like `saveConnection` only. I will think about that.

File: web/src/client/network/model.js (hunk 3 above)
Comment (imobachgs) [reply web/src/client/network/model.js:64]: Ouch, you are right.

**Review by joseivanlopez — APPROVED** [review]
LGTM

**Pull request merged**